drill-admin/src/main/kotlin/com/epam/drill/endpoints/plugin/PluginDispatcher.kt

```diff
@@ -79,8 +79,13 @@ class PluginDispatcher(override val kodein: Kodein) : KodeinAware {
                         null -> HttpStatusCode.BadRequest to "Plugin id is null for agent '$agentId'"
                         in plugins -> {
                             if (agentId in agentManager) {
-                                agentManager.addPluginFromLib(agentId, pluginId)
-                                HttpStatusCode.OK to "Plugin '$pluginId' was added to agent '$agentId'"
+                                val agentInfo = agentManager.byId(agentId)
+                                if (pluginId in agentInfo!!.plugins.map { it -> it.id }) {
```

Comment:
if (agentInfo.any { it.id == pluginId  } )

Reply:
if(pluginId in agentInfo.plugins)